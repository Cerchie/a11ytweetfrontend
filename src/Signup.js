import * as React from 'react'
import { useState } from 'react'

import { useHistory } from 'react-router-dom'

const Signup = ({ signupUser }) => {
    //setting init state
    const INITIAL_STATE = {
        username: '',
        password: '',
    }

    //saving form data, errors in state
    const [user, setUser] = useState(INITIAL_STATE)
    const [formErrors, setFormErrors] = useState([])
    const [saveConfirmed, setSaveConfirmed] = useState(false)
    //makes debugging easier
    console.debug(
        'SignupForm',
        'signup=',
        typeof signup,
        'formData=',
        user,
        'formErrors=',
        formErrors
    )
    //handles changes in form
    const handleChange = (e) => {
        const { name, value } = e.target
        setUser((user) => ({
            ...user,
            [name]: value,
        }))
    }
    //on Submit we change the page
    function goHome() {
        let history = useHistory()
        history.push('/gohome')
    }

    //handles form submit
    async function handleSubmit(e) {
        e.preventDefault()
        let result = await signupUser({
            user,
        })
        if (result.success) {
            goHome()
        } else {
            setFormErrors(result.errors)
            alert('signup not successful')
        }
        setUser(INITIAL_STATE)
        setSaveConfirmed(true)
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="shadow-md rounded-xl bg-blue-50 text-red-700 text-2xl p-6 m-6">
                <>
                    <h1 className=" rounded-xl bg-white text-gray-600 text-2xl p-6 m-6">
                        Set your username/password
                    </h1>
                    <label
                        htmlFor="username"
                        className=" text-red-700 text-2xl p-6 m-6"
                    >
                        Username
                    </label>
                    <input
                        name="username"
                        id="username"
                        className="form-control"
                        value={user.username}
                        onChange={handleChange}
                        className="border p-2 m-6"
                    />
                </>
                <>
                    <label
                        htmlFor="password"
                        className=" text-red-700 text-2xl p-6 m-6"
                    >
                        Password
                    </label>
                    <input
                        type="password"
                        id="password"
                        name="password"
                        className="form-control"
                        value={user.password}
                        onChange={handleChange}
                        className="border p-2 m-6"
                    />
                </>
                {saveConfirmed ? goHome() : null}
                <button
                    type="submit"
                    onSubmit={handleSubmit}
                    className="shadow-md rounded text-2xl text-white bg-green-500 p-2 "
                >
                    Submit
                </button>
            </div>
        </form>
    )
}

export default Signup
