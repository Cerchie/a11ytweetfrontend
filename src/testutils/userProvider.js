import React from 'react'
import UserContext from '../UserContext'

const demoUser = {
    username: 'testuser',
}

const UserProvider = ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
        {children}
    </UserContext.Provider>
)

export { UserProvider }
