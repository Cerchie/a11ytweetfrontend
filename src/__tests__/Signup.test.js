import { render, screen } from '@testing-library/react'
import React from 'react'
import Signup from '../Signup'
import { MemoryRouter } from 'react-router'
import { UserProvider } from '../testutils/userProvider'

test('renders comp', () => {
    render(<Signup />)
})

it('matches snapshot', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Signup />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})

it('matches snapshot when logged out', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <Signup />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})
