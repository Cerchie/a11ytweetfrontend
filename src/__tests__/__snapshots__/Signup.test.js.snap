// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="shadow-md rounded-xl bg-blue-50 text-red-700 text-2xl p-6 m-6"
    >
      <h1
        class=" rounded-xl bg-white text-gray-600 text-2xl p-6 m-6"
      >
        Set your username/password
      </h1>
      <label
        class=" text-red-700 text-2xl p-6 m-6"
        for="username"
      >
        Username
      </label>
      <input
        class="border p-2 m-6"
        id="username"
        name="username"
        value=""
      />
      <label
        class=" text-red-700 text-2xl p-6 m-6"
        for="password"
      >
        Password
      </label>
      <input
        class="border p-2 m-6"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        class="shadow-md rounded text-2xl text-white bg-green-500 p-2 "
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <form>
    <div
      class="shadow-md rounded-xl bg-blue-50 text-red-700 text-2xl p-6 m-6"
    >
      <h1
        class=" rounded-xl bg-white text-gray-600 text-2xl p-6 m-6"
      >
        Set your username/password
      </h1>
      <label
        class=" text-red-700 text-2xl p-6 m-6"
        for="username"
      >
        Username
      </label>
      <input
        class="border p-2 m-6"
        id="username"
        name="username"
        value=""
      />
      <label
        class=" text-red-700 text-2xl p-6 m-6"
        for="password"
      >
        Password
      </label>
      <input
        class="border p-2 m-6"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        class="shadow-md rounded text-2xl text-white bg-green-500 p-2 "
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
